fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -115,6 +116,7 @@
     private long stateRefreshTimeMillis = TimeUnit.SECONDS.toMillis(FDBDatabase.DEFAULT_RESOLVER_STATE_CACHE_REFRESH_SECONDS);
     private long transactionTimeoutMillis = DEFAULT_TR_TIMEOUT_MILLIS;
     private boolean runLoopProfilingEnabled;
+    private static volatile int threadsPerClientVersion = 1; //default is 1, which is basically disabled
```

Comment:
I'm somewhat of two minds about making this `static`. The `FDBDatabaseFactory` is supposed to be a singleton, so in theory, this shouldn't need to be `static`. This new parameter also isn't the only configuration parameter that doesn't really make sense to set twice--"run loop profiling", for example, which has a guard on the class being `inited`.

I _think_ we want this to be not-static and for it to be more like the other configuration parameters on the factory, including `runLoopProfilingEnabled`. But it's a bit awkward, especially if someone does create more than one of these `FDBDatabaseFactory` objects, which they're not supposed to do, but also #642.

I also suppose it doesn't hurt, but I think that making it `volatile` is also overkill, as all modifications happen in `synchronized` methods, so the thread cache should never have a stale value (I think).

Reply:
On the whole, I 1000% agree; static in this world feels pretty craptastic. But from a practical standpoint, making it _non-static_ has some weird problems. The scenario I ran into when I first wrote this up (using a non-static variable, a la PR #1260 ) was that creating two `FDBDatabaseFactory` instances would result in triggering an FDB-layer exception; there did not appear to be a good way of using an instance variable _and_ simultaneously telling the JVM to only use that one instance variable a single time. 

This static approach doesn't feel good to me--in particular, relying on `initDB` to not race between two instances seems iffy; what happens if you want to init two different versions in the same JVM? Is this static approach going to work? 

But in practice, it seems to be the only way around the immediate problem in my mind (unless I'm missing something, which would be great)

Reply:
Regarding the `volatile`, the idea was that it wasn't necessary to synchronize, but then I evolved to synchronize anyway. Looking over that, it looks like it's being synchronized over two different variables, which is bad. I'll fix and remove the `volatile`

Reply:
Right. In principle, `FDBDatabaseFactory` is supposed to be a singleton, though, so you should never have two of them, unless something (JUnit?) is creating more than one of these behind the scenes, which I think I've seen in the past.

It's definitely a little strange that it's static, but maybe just noting why it's static in a comment so that we know that it's because FDB hits a weird exception if you set this twice would be sufficient enough.

Reply:
multiple FDBDatabaseFactory instances are created during the RecordLayer testing code, unfortunately.

I added a comment to the `setStaticOptions` method to explain what it's doing, and why.